mcstatus/motd.py

```diff
@@ -0,0 +1,558 @@
+    """Collection of MOTD simplifies.
+
+    They're just removing unused formatting, unaffected colors etc.
+    """
+
+    def __init__(self, raw: list[ParsedMotdItem]) -> None:
+        self.parsed: list[ParsedMotdItem] = raw
+
+    def get_simplifies(self) -> Iterable[Callable[[ParsedMotdItem, int], None]]:
+        """Generator with all simplifies.
+
+        They must follow this signature:
+        ``(self, item: ParsedMotdItem, enumeration: int) -> None``
+
+        You can mock this method to add your simplifier,
+        or just add it as attribute to this class.
+        """
+        for item in dir(self):
+            if callable(getattr(self, item)) and not item.startswith("_") and item != "get_simplifies":
+                yield getattr(self, item)
+
+    def remove_double_items(self, item: ParsedMotdItem, enumeration: int) -> None:
+        """Remove double items.
+
+        This will check only next item, because anyway this check will run on all items.
+        """
+        try:
+            if self.parsed[enumeration + 1] == item:
+                self.parsed.pop(enumeration)
+        except IndexError:
+            pass
+
+    def remove_double_colors(self, item: ParsedMotdItem, enumeration: int) -> None:
+        """Remove double colors.
+
+        This will check only next item, because anyway this check will run on all items.
+        """
+        try:
+            if (isinstance(item, MinecraftColor) or isinstance(item, WebColor)) and (
+                isinstance(self.parsed[enumeration + 1], MinecraftColor) or isinstance(self.parsed[enumeration + 1], WebColor)
+            ):
+                self.parsed.pop(enumeration)
+        except IndexError:
+            pass
+
+    def remove_unused_formatting_before_color(self, item: ParsedMotdItem, enumeration: int) -> None:
+        """Remove unused formatting before color.
+
+        Color by default reset everything, so formatting will be unused.
+        """
+        try:
+            if isinstance(item, Formatting) and (
+                isinstance(self.parsed[enumeration + 1], MinecraftColor) or isinstance(self.parsed[enumeration + 1], WebColor)
+            ):
+                self.parsed.pop(enumeration)
+        except IndexError:
+            pass
+
+    def remove_non_text_in_the_end(self, item: ParsedMotdItem, enumeration: int) -> None:
+        """Remove non-text in the end.
+
+        Only strings do something in the end.
+        """
+        if enumeration == len(self.parsed) - 1 and not isinstance(item, str):
+            self.parsed.pop(enumeration)
+
+    def remove_false_values(self, item: ParsedMotdItem, enumeration: int) -> None:
+        """Remove values, that after wrapping to ``bool`` function, returns ``False``.
+
+        As example empty strings, empty lists, empty tuples, etc.
+
+        .. note:: If follow ``ParsedMotdItem`` type, this will affect only empty strings.
+        """
+        if not item:
+            self.parsed.pop(enumeration)
```

Comment:
I don't think we should do this, instead, let's just remove empty strings here, which is all this simplification does anyway, and it's much less confusing:

```suggestion
    def remove_empty_text(self, item: ParsedMotdItem, enumeration: int) -> None:
        """Remove empty text values."""
        if isinstance(item, str) and len(item) == 0:
            self.parsed.pop(enumeration)
```

Removing all falsey values is weird and if someone would subclass the Motd class and wanted to include something that's falsey, it would be quite surprising for them that it gets removed after simplification. Let's just do the obvious here.

Reply:
Fixed [here](https://github.com/py-mine/mcstatus/compare/26690ce403010ab0af019cf2230d246e2920602e..ec2cacf3a1ee18ca0432a5c5e54c80ee6dbc3fcb), thanks!